aineko/extras/connectors/rest_poller.py

```diff
@@ -0,0 +1,260 @@
+                success. Defaults to
+                    [200, 201, 202, 203, 204, 205, 206, 207, 208, 226].
+
+    Secrets can be injected (from environment) into the `url`, `headers`, and
+    `data` fields by passing a string with the following format:
+    `{$SECRET_NAME}`. For example, if you have a secret named `SECRET_NAME`
+    that contains the value `SECRET_VALUE`, you can inject it into the url
+    field by passing `https://example.com?secret={$SECRET_NAME}` as the url.
+    The connector will then replace `{$SECRET_NAME}` with `SECRET_VALUE` before
+    connecting to the REST endpoint.
+
+    Example usage in pipeline.yml:
+    ```yaml title="pipeline.yml"
+    pipeline:
+      nodes:
+        RestPoller:
+          class: aineko.extras.RestPoller
+          outputs:
+            - test_rest
+          node_params:
+            url: "https://example.com"
+            headers:
+              auth: "Bearer {$SECRET_NAME}"
+            data: {"Greeting": "Hello, world!"}
+    ```
+
+    By default, this node will poll the REST endpoint every 5 seconds. This can
+    be changed by setting the `poll_interval` field in `node_params`. If the
+    REST endpoint is not expected to be available at all times, it is
+    recommended to increase `poll_interval` to reduce the number of requests
+    sent to the endpoint.
+
+    By default, this node will retry connecting to the REST endpoint forever
+    if the connection fails. This can be changed by setting the `max_retries`
+    field in `node_params`. If the REST endpoint is not expected to be
+    available at all times, it is recommended to set `max_retries` to a
+    finite number to prevent the node from retrying forever.
+
+    By default, this node will retry connecting to the REST endpoint every 5
+    seconds if the connection fails. This can be changed by setting the
+    `retry_sleep` field in `node_params`. If the REST endpoint is not expected
+    to be available at all times, it is recommended to increase `retry_sleep`
+    to reduce the number of requests sent to the endpoint.
+
+    By default, this node will consider any HTTP status code in the 200s to be
+    a success code. This can be changed by setting the `success_codes` field
+    in `node_params`. If the REST endpoint returns a non-200 status code on
+    success, it is recommended to add the status code to the `success_codes`
+    list.
+
+    By default, this node will not attach headers or data to the request. This
+    can be changed by setting the `headers` and `data` fields in `node_params`.
+    If the REST endpoint requires headers or data to be sent, it is
+    you should set the `headers` and `data` fields in `node_params`.
+
+    By default, this node will timeout after 10 seconds. This can be changed
+    by setting the `timeout` field in `node_params`. If the REST endpoint is
+    expected to take a long time to respond, it is recommended to increase
+    `timeout` to prevent the node from timing out.
```

Comment:
I wonder if this text might be more concise and less repetitive if we document the behavior of the settings by documenting the settings themselves, as mentioned in my previous comment.

Reply:
I added these comments to address shawn's point about documenting default behavior. i thought it'd be useful to be as explicit about that as possible, as opposed to having the reader have to think about how the parameters might be expressed in the code.

Reply:
I agree with @Barabazs that we should make this more concise and put the details in the argument explanation itself. When I said default behavior I had a more generalized description in mind, like
`By default, the node will poll the endpoint every 5 seconds, and timeout after 10 seconds. It will retry forever if the connection fails. Any 20X response will be considered a success. No headers are added by default. `